src/components/Swap/Swap.tsx

```diff
@@ -126,6 +129,8 @@ export const Swap: React.FC<ISwap> = ({
     FROM = 'from',
     TO = 'to'
   }
+  const dispatch = useDispatch()
+  const isFetchingNewPool = useSelector(isLoadingLatestPoolsForTransaction)
```

Comment:
Both the `useDispatch` and `useSelector` hooks are connected to Redux. We strictly follow the rule that all components must be stateless. Therefore, the usage of these hooks should take place inside the container above.
![Screenshot from 2023-12-22 20-27-28](https://github.com/invariant-labs/webapp/assets/43304803/7926cdbe-11cf-4a0c-8ea2-e1fa28b2f417)
![Screenshot from 2023-12-22 20-28-11](https://github.com/invariant-labs/webapp/assets/43304803/da71a6a3-6528-441b-ae0f-5b2da9aa6bb5)

Reply:
we also do this because these components can be easily tested in storybook